src/helm/benchmark/augmentations/cleva_perturbation.py

```diff
@@ -0,0 +1,722 @@
+        """
+
+        prob: float = 0.0
+        source_class: str = ""
+        target_class: str = ""
+        preserve_gender: bool = False
+
+    def __init__(
+        self,
+        prob: float,
+        source_class: Dict[str, str],
+        target_class: Dict[str, str],
+        preserve_gender: bool = True,
+    ):
+        """Chinese person name perturbation. For implementation details, see
+        https://github.com/GEM-benchmark/NL-Augmenter/tree/main/nlaugmenter/transformations/chinese_person_named_entities_gender
+
+        Code adopted from
+        https://github.com/stanford-crfm/helm/blob/main/src/helm/benchmark/augmentations/person_name_perturbation.py
+
+        Args:
+            prob: Probability of substituting a word in the source class with
+                a word in the target class given that a substitution is
+                available.
+            source_class: The properties of the source class. The keys of the
+                dictionary should correspond to categories ("gender" only for
+                now) and the values should be the corresponding values. If
+                more than one category is provided. Case-insensitive.
+            target_class: Same as source_class, but specifies the target_class.
+            preserve_gender: If set to True, we preserve the gender when
+                mapping names of one category to those of another. If we can't
+                find the gender association for a source_word, we randomly
+                pick from one of the target names.
+        """
+        self.output_path: str = self.OUTPUT_PATH
+        Path(self.output_path).mkdir(parents=True, exist_ok=True)
+
+        # Assign parameters to instance variables
+        assert 0 <= prob <= 1
+        self.prob = prob
+
+        self.source_class: Dict[str, str] = self.lower_dictionary(source_class)
+        self.target_class: Dict[str, str] = self.lower_dictionary(target_class)
+
+        self.preserve_gender: bool = preserve_gender
+
+        target_path = os.path.join("benchmark_output", "perturbations", self.name, "chinese_name_gender.json")
+        ensure_directory_exists(os.path.dirname(target_path))
+        ensure_file_downloaded(source_url=self.SOURCE_URI, target_path=target_path)
+        with open(os.path.join(target_path), "r", encoding="utf-8") as f:
+            self.gender2name: Dict[str, List[str]] = json.load(f)
+            del self.gender2name["unknown"]
+
+            self.name2gender: Dict[str, str] = {}
+            for k in self.gender2name.keys():
+                for v in self.gender2name[k]:
+                    self.name2gender[v] = k
+
+    @property
+    def description(self) -> PerturbationDescription:
+        """Return a perturbation description for this class."""
+        source_str = ",".join([f"{k}={v}" for k, v in self.source_class.items()])
+        target_str = ",".join([f"{k}={v}" for k, v in self.target_class.items()])
```

Comment:
Why is this necessary as opposed to just passing in `self.source_class` and `self.target_class`?

Reply:
I think either way works: here we just follow the dict representation in [person name perturbation](https://github.com/stanford-crfm/helm/blob/65fdf7a838e06d07edd21ca408f9c56717ea90cb/src/helm/benchmark/augmentations/person_name_perturbation.py#L189).